version.go

```diff
@@ -345,6 +368,44 @@ func (v *Version) LessThanOrEqual(b *Version) bool {
 	return v.Compare(b) <= 0
 }
 
+// Something that can compare to a version. Similar to [VersionMatcher], but
+// different in that it implies order, not just a yes or no relationship.
+type VersionComparer interface {
+	CompareVersion(*Version) int
+}
+
+func (v *Version) IsNewerThan(c VersionComparer) bool {
+	return v.Is(NewerThan(c))
+}
+
+func (v *Version) IsAtLeast(c VersionComparer) bool {
+	return v.Is(AtLeast(c))
+}
+
+func (v *Version) IsOlderThan(c VersionComparer) bool {
```

Comment:
Newer/Older may not be the best naming as something like v1.30.1 could be *older* than v.1.29.20 when something is backported to the 1.29 branch